src/modules/roc_audio/profiler.cpp

```diff
@@ -0,0 +1,92 @@
+/*
+ * Copyright (c) 2019 Roc authors
+ *
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+
+#include "roc_audio/profiler.h"
+#include "roc_core/log.h"
+#include "roc_core/panic.h"
+#include "roc_core/shared_ptr.h"
+
+namespace roc {
+namespace audio {
+
+Profiler::Profiler(core::IAllocator& allocator,
+                   packet::channel_mask_t channels,
+                   size_t sample_rate,
+                   core::nanoseconds_t interval)
+    : rate_limiter_(interval)
+    , interval_(interval)
+    , chunk_length_(sample_rate / 100)
+    , num_chunks_((size_t)(interval / (core::Second / 100)) + 1)
```

Comment:
So we have 10ms chunks. Can we make this 10ms a constant? (local to profiler.cpp I think)

Reply:
Why are we always adding 1 to num_chunks? Maybe you need this instead? https://stackoverflow.com/questions/17944/how-to-round-up-the-result-of-integer-division

Reply:
> Why are we always adding 1 to num_chunks? Maybe you need this instead? https://stackoverflow.com/questions/17944/how-to-round-up-the-result-of-integer-division

refer to https://github.com/roc-project/roc/pull/291/files/7a802f6e966e0702a2fba6c6b4b3adef0e5fbf1b#r424274466